prow/pod-utils/gcs/upload.go

```diff
@@ -246,15 +267,36 @@ func (w *openerObjectWriter) Close() error {
 		}
 	}
 
-	err := w.writeCloser.Close()
-	w.writeCloser = nil
-	return err
+	var errs []error
+	for _, closer := range w.closers {
+		if err := closer.Close(); err != nil {
+			errs = append(errs, err)
+		}
+	}
+	w.closers = nil
+	w.writer = nil
+	return utilerrors.NewAggregate(errs)
 }
 
 func (w *openerObjectWriter) ApplyWriterOptions(opts pkgio.WriterOptions) {
+	if w.compressData() {
+		path := w.fullUploadPath()
+		ext := filepath.Ext(path)
+		mediaType := mime.TypeByExtension(ext)
+		if mediaType == "" {
+			mediaType = "text/plain; charset=utf-8"
+		}
+		opts.ContentType = &mediaType
+		ce := "gzip"
+		opts.ContentEncoding = &ce
+	}
 	w.opts = append(w.opts, opts)
 }
 
 func (w *openerObjectWriter) fullUploadPath() string {
 	return fmt.Sprintf("%s/%s", w.Bucket, w.Dest)
 }
+
+func (w *openerObjectWriter) compressData() bool {
```

Comment:
I don't think this is needed any longer, right?

Reply:
It is still needed for the unit test, I may be able to find another way to test it, but I am not sure it is worth the effort or complexity to change it.